blablacar-persistence/src/test/java/cz/fi/muni/pa165/teamred/dao/CommentDaoTest.java

```diff
@@ -0,0 +1,164 @@
+package cz.fi.muni.pa165.teamred.dao;
+
+import cz.fi.muni.pa165.teamred.PersistenceSampleApplicationContext;
+import cz.fi.muni.pa165.teamred.entity.*;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.TestExecutionListeners;
+import org.springframework.test.context.testng.AbstractTestNGSpringContextTests;
+import org.springframework.test.context.transaction.TransactionalTestExecutionListener;
+import org.springframework.transaction.annotation.Transactional;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import javax.persistence.EntityManager;
+import javax.persistence.PersistenceContext;
+
+import java.util.Calendar;
+import java.util.Date;
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+/**
+ * @author Erik Horváth
+ */
+@Transactional
+@ContextConfiguration(classes = PersistenceSampleApplicationContext.class)
+@TestExecutionListeners(TransactionalTestExecutionListener.class)
+public class CommentDaoTest extends AbstractTestNGSpringContextTests {
```

Comment:
Pri tomto teste sa mi zda poruseny princip single component testu, kedze sa tu vytvaraju viacere Dao objekty. Cielom je otestovat cistom CommentDao metody- cize pri create() otestovat ci sa v databaze vytvori comment zaznam a viem ho cez findById() vytiahnut von atd. Tento test by sa nemal starat o to, ci viem s komentom korektne do DB ulozit autora, ride... Nato by mal sluzit entity::CommentTest ak by sme to chceli testovat so single component pristupom. Tiez Entity Manager je nieco, co pouziva CommentDao, tu by sme toho Managera pouzivat nemali lebo sme "o uroven vyssie". Dobre by bol CommentDaoTest podla mna napisany keby mal ako field iba CommentDao. Ja som pri Dao teste vychadzal zo seminaru persistence-2, konkretne test classa Task03 a moj test som robil podla toho. Pozri ked tak moj pull request.

Reply:
Tiez sa mi to zdalo zle ale isiel som podla https://github.com/fi-muni/PA165/blob/master/eshop-persistence/src/test/java/cz/fi/muni/pa165/dao/ProductDaoTest.java

Reply:
Aj v tych nizsich test classach som to potom videl, ale nejdem to pisat vsade :D Ak by si mal nato iny nazor tak kludne napis a vydiskutujeme :)

Reply:
Aha, nj. Tak potom neviem ktory pristup je lepsi lebo su uplne odlisne a oba pochadzaju od nasich seminar tutorov :D celkom funny

Reply:
To porusenie single component testu mi ale tak bije do oci. Uz sa ale nemame koho opytat. Tak neviem :/

Reply:
Tam by sa mal testovat len jeden komponent, ostatne mockovat.

Reply:
jj, to hej, ale akurat minule hovoril na prednaske ze ak mas v test classe vacsi pocet mockov ze treba bud redizajnovat komponent alebo napisat test inak

Reply:
ale ak to chces ty takto, tak nechaj a ked tak sa ostatni vyjadria / opytame sa po milestone